File: c/armstrong-numbers/armstrong_numbers.c
#include "armstrong_numbers.h"

// Verifica numero de digitos
int numero_de_digitos(int num) {
    
    int numCopia = num, contDigs = 0;
    while (numCopia > 0 ){
        numCopia = numCopia / 10;
        contDigs++;
    }
    return contDigs;
}

// Checa se é um número de Armstrong
bool is_armstrong_number(int candidate) {
    
    int numCopia, aux, qntDig = 0, soma = 0;

    qntDig = numero_de_digitos(candidate);
    numCopia = candidate;
    
    while (numCopia > 0 ){
        aux = numCopia % 10;
        numCopia = numCopia / 10;
        soma = soma + pow(aux, qntDig);
    }
    return soma == candidate;
}

File: python/armstrong-numbers/armstrong_numbers.py
def dig_numbers(num):

    cont = 0
    while num > 0:
        num //= 10
        cont += 1
    return cont

def is_armstrong_number(number):

    orig_number = number
    exp = dig_numbers(number)
    sum = 0

    while number > 0:
        aux = number % 10
        sum += aux**exp
        number //= 10 

    return sum == orig_number
